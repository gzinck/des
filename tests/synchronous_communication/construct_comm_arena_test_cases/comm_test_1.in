{
	"states": {
		"all": ["q1", "q2", "q3", "q4", "q5"],
		"initial": ["q1"],
		"marked": [
		    [],
		    []
		],
		"bad": []
	},
	"events": {
		"all": ["a", "b"],
		"controllable": [
			["a"],
			["b"]
		],
		"observable": [
			["a"],
			["b"]
		]
	},
	"transitions": {
		"all": {
			"q1->a": ["q2"],
			"q1->b": ["q4"],
			"q2->b": ["q3"],
            "q4->a": ["q5"]
		},
		"bad": {}
	}
}
