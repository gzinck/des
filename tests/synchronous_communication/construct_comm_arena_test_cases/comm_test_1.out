{
    "events": {
        "all": [
            "[b, _, b]",
            "[a, a, _]",
            "[_, _, a]",
            "[_, b, _]"
        ],
        "controllable": [[
            "[b, _, b]",
            "[a, a, _]",
            "[_, _, a]",
            "[_, b, _]"
        ]],
        "observable": [[
            "[b, _, b]",
            "[a, a, _]",
            "[_, _, a]",
            "[_, b, _]"
        ]]
    },
    "states": {
        "all": [
            "(q3, q2, q4)",
            "(q4, q4, q5)",
            "(q3, q3, q5)",
            "(q2, q3, q2)",
            "(q3, q5, q3)",
            "(q3, q2, q5)",
            "(q2, q2, q2)",
            "(q3, q3, q3)",
            "(q5, q3, q3)",
            "(q1, q1, q1)",
            "(q3, q5, q5)",
            "(q5, q3, q4)",
            "(q1, q4, q2)",
            "(q4, q4, q4)",
            "(q3, q2, q3)",
            "(q3, q3, q4)",
            "(q3, q5, q4)",
            "(q5, q5, q5)",
            "(q4, q1, q4)",
            "(q1, q1, q2)",
            "(q4, q1, q5)",
            "(q5, q5, q3)",
            "(q2, q3, q1)",
            "(q1, q4, q1)",
            "(q5, q2, q4)",
            "(q2, q5, q2)",
            "(q2, q2, q1)",
            "(q5, q2, q3)",
            "(q5, q2, q5)",
            "(q5, q5, q4)",
            "(q5, q3, q5)",
            "(q4, q4, q3)",
            "(q4, q1, q3)",
            "(q2, q5, q1)"
        ],
        "bad": [],
        "initial": [
            "(q1, q1, q1)"
        ],
        "marked": [],
        "secrets": []
    },
    "transitions": {
        "all": {
            "(q1, q1, q1)->[_, _, a]": [
                "(q1, q1, q2)"
            ],
            "(q1, q1, q1)->[_, b, _]": [
                "(q1, q4, q1)"
            ],
            "(q1, q1, q1)->[a, a, _]": [
                "(q2, q2, q1)"
            ],
            "(q1, q1, q1)->[b, _, b]": [
                "(q4, q1, q4)"
            ],
            "(q1, q1, q2)->[_, b, _]": [
                "(q1, q4, q2)"
            ],
            "(q1, q1, q2)->[a, a, _]": [
                "(q2, q2, q2)"
            ],
            "(q1, q1, q2)->[b, _, b]": [
                "(q4, q1, q3)"
            ],
            "(q1, q4, q1)->[_, _, a]": [
                "(q1, q4, q2)"
            ],
            "(q1, q4, q1)->[a, a, _]": [
                "(q2, q5, q1)"
            ],
            "(q1, q4, q1)->[b, _, b]": [
                "(q4, q4, q4)"
            ],
            "(q1, q4, q2)->[a, a, _]": [
                "(q2, q5, q2)"
            ],
            "(q1, q4, q2)->[b, _, b]": [
                "(q4, q4, q3)"
            ],
            "(q2, q2, q1)->[_, _, a]": [
                "(q2, q2, q2)"
            ],
            "(q2, q2, q1)->[_, b, _]": [
                "(q2, q3, q1)"
            ],
            "(q2, q2, q1)->[b, _, b]": [
                "(q3, q2, q4)"
            ],
            "(q2, q2, q2)->[_, b, _]": [
                "(q2, q3, q2)"
            ],
            "(q2, q2, q2)->[b, _, b]": [
                "(q3, q2, q3)"
            ],
            "(q2, q3, q1)->[_, _, a]": [
                "(q2, q3, q2)"
            ],
            "(q2, q3, q1)->[b, _, b]": [
                "(q3, q3, q4)"
            ],
            "(q2, q3, q2)->[b, _, b]": [
                "(q3, q3, q3)"
            ],
            "(q2, q5, q1)->[_, _, a]": [
                "(q2, q5, q2)"
            ],
            "(q2, q5, q1)->[b, _, b]": [
                "(q3, q5, q4)"
            ],
            "(q2, q5, q2)->[b, _, b]": [
                "(q3, q5, q3)"
            ],
            "(q3, q2, q3)->[_, b, _]": [
                "(q3, q3, q3)"
            ],
            "(q3, q2, q4)->[_, _, a]": [
                "(q3, q2, q5)"
            ],
            "(q3, q2, q4)->[_, b, _]": [
                "(q3, q3, q4)"
            ],
            "(q3, q2, q5)->[_, b, _]": [
				"(q3, q3, q5)"
            ],
            "(q3, q3, q4)->[_, _, a]": [
                "(q3, q3, q5)"
            ],
            "(q3, q5, q4)->[_, _, a]": [
                "(q3, q5, q5)"
            ],
            "(q4, q1, q3)->[_, b, _]": [
                "(q4, q4, q3)"
            ],
            "(q4, q1, q3)->[a, a, _]": [
                "(q5, q2, q3)"
            ],
            "(q4, q1, q4)->[_, _, a]": [
                "(q4, q1, q5)"
            ],
            "(q4, q1, q4)->[_, b, _]": [
                "(q4, q4, q4)"
            ],
            "(q4, q1, q4)->[a, a, _]": [
                "(q5, q2, q4)"
            ],
            "(q4, q1, q5)->[_, b, _]": [
                "(q4, q4, q5)"
            ],
            "(q4, q1, q5)->[a, a, _]": [
                "(q5, q2, q5)"
            ],
            "(q4, q4, q3)->[a, a, _]": [
                "(q5, q5, q3)"
            ],
            "(q4, q4, q4)->[_, _, a]": [
                "(q4, q4, q5)"
            ],
            "(q4, q4, q4)->[a, a, _]": [
                "(q5, q5, q4)"
            ],
            "(q4, q4, q5)->[a, a, _]": [
                "(q5, q5, q5)"
            ],
            "(q5, q2, q3)->[_, b, _]": [
                "(q5, q3, q3)"
            ],
            "(q5, q2, q4)->[_, _, a]": [
                "(q5, q2, q5)"
            ],
            "(q5, q2, q4)->[_, b, _]": [
                "(q5, q3, q4)"
            ],
            "(q5, q2, q5)->[_, b, _]": [
                "(q5, q3, q5)"
            ],
            "(q5, q3, q4)->[_, _, a]": [
                "(q5, q3, q5)"
            ],
            "(q5, q5, q4)->[_, _, a]": [
                "(q5, q5, q5)"
            ]
        },
        "bad": {}
    }
}
